lib/Server.js

```diff
@@ -2258,6 +2258,11 @@ class Server {
         this.middleware = null;
       }
     }
+
+    // We add listeners to signals when creating a new Server instance
+    // So ensure they are removed to prevent EventEmitter memory leak warnings
+    process.removeAllListeners("SIGINT");
+    process.removeAllListeners("SIGTERM");
```

Comment:
We should not remove all of them just only our, move our close function above the file and use `removeListener`

Reply:
above where?

Reply:
I.e. just move it in function above the Server class and remove this listener in close